core/src/main/java/org/elasticsearch/action/support/replication/ReplicationRequest.java

```diff
@@ -41,12 +41,11 @@
 
     public static final TimeValue DEFAULT_TIMEOUT = new TimeValue(1, TimeUnit.MINUTES);
 
-    ShardId internalShardId;
+    ShardRouting internalShardRouting;
```

Comment:
the `AsyncPrimaryAction` is using this now.  We can remove this but then we need to resolve the primary twice. I don't have a real preference, since it is all local now.